src/adapters/lokijs/worker/performJoins/index.js

```diff
@@ -0,0 +1,35 @@
+// @flow
+
+import type { LokiQuery, LokiJoin, LokiRawQuery } from '../encodeQuery'
+import type { DirtyRaw } from '../../../../RawRecord'
+
+type QueryPerformer = (join: LokiJoin) => DirtyRaw[]
+
+function performJoinsImpl(query: LokiRawQuery, performer: QueryPerformer): LokiRawQuery {
+  if (!query) {
+    return query
+  } else if (query.$join) {
+    const join: LokiJoin = query.$join
+    const records = performer(join)
```

Comment:
yep, this is to make this a pure function, so that it's easy to test this logic without mocking or testing at the whole database level. Alternatively, we'd still need to pass `loki: Loki` as parameter, so there's no difference in the end, and this one's cleaner

Reply:
Ok, thanks :)